src/libappimage/utils/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+add_library(
+    appimage_utils OBJECT
+    MagicBytesChecker.cpp
+    ElfFile.cpp
+    HashLib.cpp
+    UrlEncoder.cpp
+    IconHandle.cpp
+)
+
+# Generic module configuration
+configure_libappimage_module(appimage_utils)
+target_include_directories(
+    appimage_utils
+    PRIVATE $<TARGET_PROPERTY:Boost::filesystem,INTERFACE_INCLUDE_DIRECTORIES>
```

Comment:
Can just use `target_link_libraries` when making this a static lib.

Reply:
No, cmake 3.8 doesn't support it

Reply:
For static libs, `target_link_libraries` should do the job since CMake 3.0 at least.

We aren't bound to CMake 3.8.